pkg/providers/snow/snow.go

```diff
@@ -176,19 +183,47 @@ func (p *snowProvider) GetInfrastructureBundle(clusterSpec *cluster.Spec) *types
 }
 
 func (p *snowProvider) DatacenterConfig() providers.DatacenterConfig {
-	return nil
+	return p.datacenterConfig
 }
 
 func (p *snowProvider) DatacenterResourceType() string {
-	return ""
+	return snowDatacenterResourceType
 }
 
 func (p *snowProvider) MachineResourceType() string {
-	return ""
+	return snowMachineResourceType
 }
 
 func (p *snowProvider) MachineConfigs() []providers.MachineConfig {
-	return nil
+	configs := make(map[string]providers.MachineConfig, len(p.machineConfigs))
+	controlPlaneMachineName := p.clusterConfig.Spec.ControlPlaneConfiguration.MachineGroupRef.Name
+	p.machineConfigs[controlPlaneMachineName].Annotations = map[string]string{p.clusterConfig.ControlPlaneAnnotation(): "true"}
+	if p.clusterConfig.IsManaged() {
+		p.machineConfigs[controlPlaneMachineName].SetManagedBy(p.clusterConfig.ManagedBy())
```

Comment:
Why don't we set this in one of the "setup" methods?
Same for the other machine configs